test/libsolidity/syntaxTests/scoping/for_single_statement.sol

```diff
@@ -0,0 +1,9 @@
+contract C
+{
+	function f(uint x) public pure {
+		for (uint data; data < x;)
+			uint data2 = 1;
```

Comment:
Should this be valid syntax wise in the first place?

Reply:
I'm not sure, but this is not valid in C/C++, so that may be a good indication we don't want it here.

Reply:
With which options? With default g++ it compiles for me.

Reply:
Ah, it is debatable then, C++ allows it, C doesn't.

Reply:
Ah, right